Example-Crypto-REXX/RXCRYPT1.rexx

```diff
@@ -0,0 +1,2453 @@
+        if CryptoAttr.i = ASSIGNED_DOMAINS_PROP then
+          do
+            if getassignedDomains(ArrayResponse) <> 0 then
+              return fatalError('failed to parse assigned domains')
+          end
+        else if CryptoAttr.i = ASSIGNED_CRYPTOS_PROP then
+          do
+            if getassignedCryptos(ArrayResponse) <> 0 then
+              return fatalError('failed to parse assigned cryptos')
+          end
+        else
+          return fatalError('Requested attribute not recognized')
+      end /* complex property with nested content */
+
+end /* i, attribute count */
+say '  '
+say 'Finished processing content for LPAR '||LPARName
+say '  '
+return 0
+
+/*******************************************************/
+/* Function: getAssignedCryptos                        */
+/*                                                     */
+/* Retrieve content returned from the assigned cryptos */
+/* value, which is an array of zero or more objects.   */
+/* Store the specific content into REXXWRT stem        */
+/*                                                     */
+/* Return 0 if successful, otherwise return -1         */
+/*******************************************************/
+getAssignedCryptos:
+ ArrayResponse = arg(1)
+
+ keyType = 'assigned-cryptos'
+
+ /* First assess if this property was even found in the response */
+ if foundJSONValue(ArrayResponse) = FALSE then
+   do
+     say '** Unable to locate '||keyType||' in the response **'
+     return 0
+   end
+
+ /****************************************************************/
+ /* Determine the number of items in the array                   */
+ /****************************************************************/
+ Cryptos = JSON_getArrayDim(ArrayResponse)
+ if Cryptos = -1 then
+    return fatalError( '** Unable to retrieve cryptos **' )
+ else if Cryptos = 0 then
+    say '** Empty list of cryptos returned **'
+
+ /****************************************************************/
+ /* Traverse the array to populate Cryptos list                  */
+ /****************************************************************/
+ If Cryptos > 0 then do
+   say 'Processing information for '||Cryptos||' cryptos'
+   say 'Number , Activation Type '
+   outLC = outLC + 1
+   REXXWRT.outLC = ',,,'||keyType||' (Number & Activation Type Array)'
+   outLC = outLC + 1
+ drop CryList.
+ CryList.0 = Cryptos
+ do j = 1 to Cryptos
+    nextEntryHandle = JSON_getArrayEntry(ArrayResponse,j-1)
+    CryList.j.num=JSON_findValue2(nextEntryHandle,"number")
+    CryList.j.type=JSON_findValue(nextEntryHandle,"activation-type",,
+                        HWTJ_STRING_TYPE)
+
+    say CryList.j.num || ' , ' || Crylist.j.type
+                          
+    if j=1 then
+      REXXWRT.outLC = ',,,'||CryList.j.num||','||CryList.j.type
+    else
+      REXXWRT.outLC = REXXWRT.outLC||','||CryList.j.num||','||CryList.j.type
+
+ end /* array entries */
+ end /* Cryptos  */
```

Comment:
would you fix the indentation here, you can use ',' to continue a line on the line if you run past col 80
i.e.  how you have: CryList.j.type=JSON_findValue(nextEntryHandle,"activation-type",,
                        HWTJ_STRING_TYPE)

the two ',,' -> one is a comma for a parm and the second is a line continuation

Reply:
Done